virtio-devices/src/block.rs

```diff
@@ -258,18 +292,21 @@ impl EpollHelperHandler for BlockEpollHandler {
                     return true;
                 }
 
-                match self.process_queue_submit() {
-                    Ok(needs_notification) => {
-                        if needs_notification {
-                            if let Err(e) = self.signal_used_queue() {
-                                error!("Failed to signal used queue: {:?}", e);
-                                return true;
+                // Process the queue only when the 'rate_limiter' is not blocked
+                if !self.rate_limiter.is_blocked() {
+                    match self.process_queue_submit() {
+                        Ok(needs_notification) => {
```

Comment:
You could write this as `Ok(true)`

Reply:
If we explicitly match for `Ok(true)`, we will need to add another path to handle `Ok(false)` or `_` that does nothing, which seems to more counter-intuitive (than the current implementation). I am fine either way if you have a preference. :)